src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -483,6 +486,7 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
             }
         }
 
+        // n.b. this will extend the array if necessary and fill in the gaps with undefined
```

Comment:
It will actually make it a [sparse array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Indexed_collections#sparse_arrays) with empty slots that are skipped by iteration methods.

Which gets me thinking, maybe callers can hook into that...

Reply:
> maybe callers can hook into that...

That is indeed the case; see https://github.com/element-hq/element-call/pull/2646#discussion_r1771593805 which makes the updated typehints of this PR unnecessary (though the comments & updated unit test remain useful).

Reply:
Well, TypeScript doesn't seem to model sparse arrays at all. So, as far as I can see, there is no compile time hint that a sparse array can contain empty values.

I don't find this very satisfactory. /me ponders

Reply:
There is [`--noUncheckedIndexedAccess`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#checked-indexed-accesses---nouncheckedindexedaccess), but that would require updating many other parts of the codebase to comply with it.